### ISSUE TITLE
Ability to define the default global policy chain using environment variables

### ISSUE THREAD

**Issue by nmasse-itix** [3scale/apicast#608]
When developing policies that are meant to be used as global policy, it can be tedious since it is required to write a lua module that defines the global policy chain as explained here : 
https://github.com/3scale/apicast/blob/master/examples/policy_chain/configuration.lua

It would be nice to be able to define the global policy chain using environment variables. 

This would allow a customer to choose his global policy chain at runtime (instead of build time).

**Comment by mikz** [comment]
ENV variable is just a form of a configuration file. So I'd say the recommended approach should be to mount a configuration from ConfigMap. 

I don't see how we could define so complicated structure in ENV variables.

**Comment by nmasse-itix** [comment]
I agree that either ConfigMap or Environment Variables could be used to configure APIcast and I have nothing against ConfigMaps. :)

My main issue is that currently to change the global policy chain, someone has to develop a small piece of code in LUA. Especially, LUA code in ConfigMap is hard to maintain. 

So I would rephrase this issue as "The global policy chain needs to be defined by configuration, not code".

Namely, we could define the global policy chain as a JSON file, deployed on the file system using ConfigMap. 

Would it be more suitable ?

**Comment by mikz** [comment]
@nmasse-itix yes that is more suitable and can actually be prototyped right now. 

This is internal API for now, so is not meant for customer use, but an image can define what lua files to load like https://github.com/3scale/apicast-cloud-hosted/pull/1/files#diff-70ad260657019e0d5afd56229a1c8184

And then the lua file like https://github.com/3scale/apicast-cloud-hosted/pull/1/files#diff-ffecfd7a78dd22dd10038f4a0ebc0dfc
could get the JSON file from some ENV, load it and update the policy chain.

We haven't done it because it is hard to define how it should interact with the default global chain. 
APIcast requires some internal policies for it to work and if some JSON is defined it is not clear if it should replace it, extend it, add policies before or after. Those operations are hard to define as JSON. 

So we could totally prepare this as an example and see how it works. If it is good enough experience then try to merge it to the core.

**Comment by nmasse-itix** [comment]
Thanks @mikz for the hint ! Il will have a look at it.
